packages/contracts/src/dollar/mocks/MockChainLinkFeed.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+import {AggregatorV3Interface} from "@chainlink/interfaces/AggregatorV3Interface.sol";
+
+contract MockChainLinkFeed is AggregatorV3Interface {
+    uint80 roundId;
+    int256 answer;
+    uint256 startedAt;
+    uint256 updatedAt;
+    uint80 answeredInRound;
+
+    constructor() {
+        roundId = 0;
+        answer = 0;
+        startedAt = block.timestamp;
+        updatedAt = block.timestamp;
+        answeredInRound = 0;
+    }
+
+    function decimals() external pure override returns (uint8) {
+        return 18;
+    }
+
+    function description() external pure override returns (string memory) {
+        return "mock ChainLink data feed";
+    }
+
+    function version() external pure override returns (uint256) {
+        return 1;
+    }
+
+    function getRoundData(
+        uint80
```

Comment:
why we should cast the value to uint80?

Reply:
I'll check it

Reply:
yeah i see now this is part of the implementation